libs/arrow-spi/build.gradle

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+testingConventions.enabled = false
+dependencies {
+  implementation project(':libs:opensearch-core')
+  api "org.apache.arrow:arrow-vector:${versions.arrow}"
+  api "org.apache.arrow:arrow-format:${versions.arrow}"
+  api "org.apache.arrow:arrow-memory-core:${versions.arrow}"
+  runtimeOnly "org.apache.arrow:arrow-memory-netty-buffer-patch:${versions.arrow}"
+  runtimeOnly "org.apache.arrow:arrow-memory-netty:${versions.arrow}"
+  runtimeOnly "io.netty:netty-buffer:${versions.netty}"
+  runtimeOnly "io.netty:netty-common:${versions.netty}"
+
+  runtimeOnly "com.google.flatbuffers:flatbuffers-java:${versions.flatbuffers}"
+  implementation "org.slf4j:slf4j-api:${versions.slf4j}"
```

Comment:
```suggestion
  api "org.slf4j:slf4j-api:${versions.slf4j}"
```

Reply:
curious why we want to make them api dependency?

Reply:
I think any implementation of this SPI would require SLF4J, right?

Reply:
that's right but only at runtime and not at compileTime classpath. Like `arrow-vector` is a api dependency since I'm exposing them as part of public APIs whereas `slf4j` is used internally by `arrow-vector` and doesn't have to be part of compile classpath of server. Please correct me if my there is a gap in my understanding, thanks.

Reply:
Makes sense. Verified that its not needed at compile time at all for `libs:arrow-spi`, changed it to `runtimeOnly`. Thanks.